ssl/t1_lib.c

```diff
@@ -2897,6 +2897,24 @@ static int sig_cb(const char *elem, int len, void *arg)
      * in the table.
      */
     if (p == NULL) {
+        /* First check the providers */
+        SSL_CTX *ctx_helper = OPENSSL_zalloc(sizeof(*ctx_helper));
+        /* Load provider sigalgs */
+        if (!ctx_helper || !ssl_load_sigalgs(ctx_helper)) {
```

Comment:
`ssl_load_sigalgs` reads `ctx->libctx` directly and `ctx->propq` indirectly (via `discover_provider_sig_algs`->`add_provider_sigalgs`) - neither of which have been set here. This is potentially quite problematic since `tls1_set_sigalgs_list` isn't passed this information either (although all callers of `tls1_set_sigalgs_list` do have it)

Probably a better solution is just to ensure the real SSL_CTX is passed down the call chain, rather than constructing a dummy one here.

Reply:
I pushed an update that potentially addresses this and detailed it in a comment below. If you could give feedback on the "FEEDBACK QUESTION" I posed in that update.